dodo.py

```diff
@@ -1,107 +1,143 @@
-import functools
+import itertools
+import os
 import pathlib
+import shlex
 
-from doit.action import CmdAction as _CmdAction
+from doit.action import CmdAction
 
-CmdAction = functools.partial(_CmdAction, shell=False)
 
 HERE = pathlib.Path(__file__).parent
+PACKAGE_NAME = "light_the_torch"
 
+CI = os.environ.get("CI") == "1"
 
-def task_setup(pip="pip"):
-    return dict(
+DOIT_CONFIG = dict(
+    verbosity=2,
+    backend="json",
+)
+
+
+def do(*cmd):
+    if len(cmd) == 1:
+        cmd = cmd[0]
+    if isinstance(cmd, str):
+        cmd = shlex.split(cmd)
+    return CmdAction(cmd, shell=False, cwd=HERE)
+
+
+def task_install(pip="python -m pip"):
+    """Installs all development requirements and light-the-torch in development mode"""
+    yield dict(
+        name="requirements",
+        actions=[do(f"{pip} install -r requirements-dev.txt")],
+    )
+    yield dict(
+        name="install",
         actions=[
-            [pip, "install", "-r", HERE / "requirements-dev.txt"],
-            [pip, "install", "-e", HERE],
+            do(f"{pip} install -e ."),
         ],
     )
 
 
-def task_dev_env():
+def task_setup():
+    """Sets up a development environment for light-the-torch"""
     dev_env = HERE / ".venv"
     return dict(
         actions=[
-            ["rm", "-rf", dev_env],
-            ["virtualenv", dev_env, "--prompt=(light-the-torch-dev) "],
-            *task_setup(dev_env / "bin" / "pip")["actions"],
-            [
-                "echo",
-                f"run `source {dev_env / 'bin' / 'activate'}` the virtual environment",
-            ],
+            do(f"virtualenv {dev_env} --prompt='(light-the-torch-dev) '"),
+            *itertools.chain.from_iterable(
+                sub_task["actions"]
+                for sub_task in task_install(dev_env / "bin" / "pip")
```

Comment:
i'd avoid naming task factories with `task_` if you must use tasks factories then maybe `_task_*`  would prevent future collisions.

Reply:
Good point. I'll factor it out.